src/stp/PlayChecker.cpp

```diff
@@ -20,4 +20,13 @@ std::vector<Play*> PlayChecker::getValidPlays() noexcept {
 void PlayChecker::update(world_new::World* world) noexcept { this->world = world; }
 
 void PlayChecker::setPlays(std::vector<std::unique_ptr<Play>>& plays) noexcept { this->allPlays = &plays; }
+
+Play* PlayChecker::getPlayForName(const std::string& playName) const noexcept {
+    for (auto& each : *allPlays) {
+        if(each->getName() == playName) {
+            return &*each;
+        }
+    }
+    return nullptr;
```

Comment:
std::find_if?

Reply:
```suggestion
    auto found = std::find_if(allPlays->begin(), allPlays->end(), [&](auto& play) { return play->getName() == playName; });
   if (found == allPlays->end()) { 
       return nullptr; 
    } else return found->get();
}
```

Reply:
Actually since we're looking up plays by name wouldn't it be better to have an O(1) operation? (std::unordered_map)

Reply:
True, but this only happens when there are no valid plays which is not that often. I do agree that an unordered map would make more sense if we would want to lookup by name more often